qiskit_experiments/library/driven_freq_tuning/coefficient.py

```diff
@@ -0,0 +1,278 @@
+            if amp > 0:
+                fit = self.positive_coeffs()
+            else:
+                fit = self.negative_coeffs()
+            # Solve for inflection points by computing the point where derivative becomes zero.
+            solutions = np.roots([deriv * coeff for deriv, coeff in zip((3, 2, 1), fit)])
+            inflection_points = solutions[
+                (solutions.imag == 0) & (np.sign(solutions) == np.sign(amp))
+            ]
+            if len(inflection_points) > 0:
+                # When multiple inflection points are found, use the most outer one.
+                # There could be a small inflection point around amp=0,
+                # when the first order term is significant.
+                amp = min([amp, max(inflection_points, key=abs)], key=abs)
+            trim_amps.append(amp)
+        return tuple(self.convert_amp_to_freq(np.asarray(trim_amps)))
+
+    def __str__(self):
+        # Short representation for dataframe
+        return "StarkCoefficients(...)"
+
+    def __eq__(self, other):
+        return all(
+            [
+                self.pos_coef_o1 == other.pos_coef_o1,
+                self.pos_coef_o2 == other.pos_coef_o2,
+                self.pos_coef_o3 == other.pos_coef_o3,
+                self.neg_coef_o1 == other.neg_coef_o1,
+                self.neg_coef_o2 == other.neg_coef_o2,
+                self.neg_coef_o3 == other.neg_coef_o3,
+            ]
+        )
+
+    def __json_encode__(self) -> dict[str, Any]:
+        return {
+            "class": "StarkCoefficients",
+            "data": {
+                "pos_coef_o1": self.pos_coef_o1,
+                "pos_coef_o2": self.pos_coef_o2,
+                "pos_coef_o3": self.pos_coef_o3,
+                "neg_coef_o1": self.neg_coef_o1,
+                "neg_coef_o2": self.neg_coef_o2,
+                "neg_coef_o3": self.neg_coef_o3,
+                "offset": self.offset,
+            },
+        }
+
+    @classmethod
+    def __json_decode__(cls, value: dict[str, Any]) -> "StarkCoefficients":
+        if not value.get("class", None) == "StarkCoefficients":
+            raise ValueError("JSON decoded value for StarkCoefficients is not valid class type.")
+        return StarkCoefficients(**value.get("data", {}))
+
+
+def retrieve_coefficients_from_service(
+    service: IBMExperimentService,
+    backend_name: str,
+    qubit: int,
+) -> StarkCoefficients:
+    """Retrieve StarkCoefficients object from experiment service.
+
+    Args:
+        service: IBM Experiment service instance interfacing with result database.
+        backend_name: Name of target backend.
+        qubit: Index of qubit.
+
+    Returns:
+        StarkCoefficients object.
+
+    Raises:
+        RuntimeError: When stark_coefficients entry doesn't exist in the service.
+    """
+    try:
+        if isinstance(qubit, (list, tuple)) and len(qubit) == 1:
+            qubit = qubit[0]
+        retrieved = service.analysis_results(
```

Comment:
Should this put `limit=1`? I noticed that the qiskit-ibm-experiment has a default limit of 10 because when I first tried to test the new code I kept getting errors about `coefficient_util` not existing. I realized that this was from results from my previous testing still being in the most recent 10 results that were retrieved.

Reply:
Thanks for finding this. Likely this is an edge case but worth adding the limit. Done in [aaee397](https://github.com/Qiskit-Extensions/qiskit-experiments/pull/1236/commits/aaee397ce70b12f72d8e6a27879bf21c7e801d7e)